src/feature/Navbar/components/NavItem/NavItem.tsx

```diff
@@ -0,0 +1,39 @@
+import { NavLink } from "react-router-dom";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import styles from "./NavItem.module.scss";
+
+type NavItemProps = {
+  children: string;
+  icon: IconProp;
+  destination: string;
+  onClick: () => void;
+};
+
+export const NavItem = ({
+  children,
+  icon,
+  destination,
+  onClick,
+}: NavItemProps) => {
+  return (
+    <li>
+      <NavLink
+        to={destination}
+        className={({ isActive, isPending }) =>
+          isPending
```

Comment:
Piekło ternary operatorów. Proponuję wyprowadzić to albo do funkcji poza JSX, a najlepiej użyłbym paczki `classnames` żeby poradzić sobie z tymi warunkami.